### PR TITLE
Added basic error handling

### PR DESCRIPTION
**Pull request by oliverisaac**
Set it so it will return an OS level exit code when there are errors.

### CODE REVIEW

File: gucci.go [hunk 1]
```diff
@@ -47,10 +47,13 @@ func main() {
 		tplPath := c.Args().First()
 		vars, err := loadVariables(c)
 		if err != nil {
-			return err
+			return cli.NewExitError(err, 14)
```
Comment (noqcks) [inline gucci.go]: What the reason for exit code 14 here?

Same questions for 32 I guess

**Comment by noqcks** [comment]
Hey @oliverisaac thanks for the PR!

Can you make sure tests are passing

File: gucci.go (hunk 1 above)
Comment (oliverisaac) [reply gucci.go]: Yeah, I picked random numbers. I switched them to 1 as if we're going to do different error codes it'd be nice to not have there be just 2 options. (e.g. return a cli.NewExitError further down in the code.)

**Comment by oliverisaac** [comment]
Tests are passing now. I created a `runWithExitCode(command, exitCode)` function so one can specify an eixt code and then made `Run(command)` just specify an exit code of 0.

**Comment by noqcks** [comment]
Great thank you! 

I'll cut a release after merging this

**Pull request merged**